### PR TITLE
Fix constructs, add class name, support getters, and param with default values

### PR DESCRIPTION
**Pull request by christophehurpeau**

### CODE REVIEW

File: test/fixtures/actual.js [hunk 1]
```diff
@@ -27,9 +40,12 @@ class A {
         arg5: Array<string>,
         arg6: Class<SomeOtherClass>,
         arg7: Function,
-        arg7: Map<string, number>,
-        arg8: Set<string>,
-        arg9: Promise<string>
+        arg8: Map<string, number>,
+        arg9: Set<string>,
+        arg10: Promise<string>,
+        arg11?: int,
```
Comment (cyrilf) [inline test/fixtures/actual.js:47]: Quite not sure about this: `?`. Typo?

Comment (christophehurpeau) [inline test/fixtures/actual.js:47]: It's optional parameters : http://flowtype.org/docs/functions.html

Comment (christophehurpeau) [inline test/fixtures/actual.js:47]: `of the form x?, indicating an optional parameter`

Comment (cyrilf) [inline test/fixtures/actual.js:47]: Ok, thanks. Got it!
Didn't know about flowtype, so I have no idea what am I doing here! :wink:

**Comment by motiz88** [comment]
Fantastic, thanks!

**Pull request merged**